test/harness.c

```diff
@@ -63,10 +64,15 @@ static testresult_t run_test_child(const char *self, test_index_entry *entry)
 	static int inproc_warn = 0;
 
 #ifdef HAVE_OS_EXEC
-	int status;
-
-	if (os_exec(&status, NULL, self, entry->name, (char *)NULL))
-		return status;
+	int status, abnormal_exit;
+
+	if (os_exec(&status, &abnormal_exit, NULL, self, entry->name, (char *)NULL))
+	{
+		if (abnormal_exit)
```

Comment:
IMO this should be use the ternary operator

Reply:
Can do.